### PR TITLE
GPTQ feature enhance

### PR DESCRIPTION
**Pull request by YIYANGCAI**
## Type of Change

Enhance

## Description

Enhance GPTQ function to be more flexible and align some new function of INC

## How has this PR been tested?

PreCI

## Dependency Change?

None

### CODE REVIEW

File: test/adaptor/pytorch_adaptor/test_weight_only_adaptor.py [hunk 1]
```diff
@@ -268,8 +268,8 @@ def __iter__(self):
                 'gptq_args':{'percdamp': 0.01},
             },
         )
-        dataloader = gptq_inc_loader()
         # import pdb;pdb.set_trace()
```
Comment (hshen14) [inline test/adaptor/pytorch_adaptor/test_weight_only_adaptor.py:271]: remove unused code

File: test/adaptor/pytorch_adaptor/test_weight_only_adaptor.py [hunk 2]
```diff
@@ -69,167 +69,167 @@ def tearDownClass(self):
...
-        )
-        input = torch.ones([1, 10], dtype=torch.long)
-        out1 = q_model(input)
-        q_model.export_compressed_model()
-        out2 = q_model(input)
-        # no idea about the gap at 1e-08, use allclose instead of out1==out2
-        self.assertTrue(torch.allclose(out1[0], out2[0], atol=1e-05))
-        self.assertTrue(isinstance(q_model.model.transformer.h[0].mlp.fc_in, WeightOnlyLinear))
-        self.assertTrue(isinstance(q_model.model.lm_head, torch.nn.Linear))
+    # def test_RTN_quant(self):
```
Comment (hshen14) [inline test/adaptor/pytorch_adaptor/test_weight_only_adaptor.py:72]: why comment the tests?

File: neural_compressor/adaptor/pytorch.py [hunk 3]
```diff
@@ -4559,21 +4560,44 @@ def rtn_quantize(self, model, tune_cfg):
         return model
 
     def gptq_quantize(self, model, tune_cfg, dataloader):
+        import pdb;pdb.set_trace()
```
Comment (hshen14) [inline neural_compressor/adaptor/pytorch.py:4563]: remove debugging code

File: neural_compressor/adaptor/pytorch.py [hunk 4]
```diff
@@ -4559,21 +4560,44 @@ def rtn_quantize(self, model, tune_cfg):
...
+                'group_size': 128,
+                'sym': False,
+                'percdamp': 0.01,
+                'actorder': True
+            },
+            "layer_name_2": {
+                'wbits': 4,
+                'group_size': 128,
+                'sym': False,
+                'percdamp': 0.01,
```
Comment (hshen14) [inline neural_compressor/adaptor/pytorch.py:4581]: what does it mean?

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_weight_only/datautils.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
+import numpy as np
+import torch
+import datasets
+
+cache_dir = "/home/wenhuach/.cache/"
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_weight_only/datautils.py:5]: ~~

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_weight_only/datautils.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+    return trainloader, valenc 
+
+def get_pile(nsamples, seed, seqlen, model):
+    from datasets import load_dataset
+
+    """
+    traindata = load_dataset(
+        'NeelNanda/pile-10k', split='train',
+        cache_dir="/home/wenhuach/.cache/huggingface/"
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/ptq_weight_only/datautils.py:113]: same issue

File: neural_compressor/adaptor/pytorch.py (hunk 4 above)
Comment (wenhuach21) [reply neural_compressor/adaptor/pytorch.py:4585]: I guess it is to  improve the robustness of the inversing of matrixes by adding this value to the diag

**Review by hshen14 — APPROVED** [review]

**Review by wenhuach21 — APPROVED** [review]

**Pull request merged**